File: jimmer-example/spring-boot-with-jimmer/src/main/java/org/jimmer/domain/Category.java
package org.jimmer.domain;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.Key;
import org.babyfish.jimmer.sql.OneToMany;
import org.babyfish.jimmer.sql.Table;

import java.util.List;

/**
 * @author jhlz
 * @version x.x.x
 */
@Entity
@Table(name = "tb_category")
public interface Category extends ID {

    @Key
    String name();

    @OneToMany(mappedBy = "category")
    List<Product> products();

}

File: jimmer-example/spring-boot-with-jimmer-kt/src/main/kotlin/org/jimmer/domain/Category.kt
package org.jimmer.domain

import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

/**
 * @author jhlz
 * @version x.x.x
 */
@Entity
interface Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long
    @IdView
    val parentId: Long?
    val name: String
    val code: String
    val createTime: LocalDateTime?


    /**
     * 自关联的树
     */
    @ManyToOne
    @JoinColumn( foreignKeyType = ForeignKeyType.FAKE)
    val parent: Category?
    @OneToMany(mappedBy = "parent")
    val childs: List<Category>
}